BookReader/BookReader.js

```diff
@@ -3873,16 +3890,50 @@ BookReader.prototype.navigationIsVisible = function() {
     return tooltop == 0;
 };
 
+/**
+ * Main controller that toggles navigation.
+ * Defaults to SHOW the navigation chrome
+ */
+BookReader.prototype.toggleNavigation = function brToggleNavigation(hide) {
+    var animationLength = this.constNavAnimationDuration;
+    var animationType = 'linear';
+    var resizePageContainer = function resizePageContainer () {
+        /* main page container fills whole container */
+        if (this.constMode2up !== this.mode) {
+            var fillContainerWithBook = true;
+            this.resizeBRcontainer(fillContainerWithBook);
+        }
+        this.trigger(BookReader.eventNames.navToggled);
+    }.bind(this);
+
+    var toolbarHeight = 0;
+    var navbarHeight = 0;
+    if (hide) {
+        toolbarHeight = this.getToolBarHeight() * -1;
+        navbarHeight = this.getNavHeight() * -1;
+    }
+
+    this.refs.$BRtoolbar.animate(
+        { top: toolbarHeight },
+        animationLength,
+        animationType,
+        resizePageContainer
+    );
+    this.refs.$BRnav.animate(
+        { bottom: navbarHeight },
+        animationLength,
+        animationType,
+        resizePageContainer
+    );
+};
 /**
  * Hide navigation elements, if visible
  */
 BookReader.prototype.hideNavigation = function() {
     // Check if navigation is showing
     if (this.navigationIsVisible()) {
-        var toolbarHeight = this.getToolBarHeight();
-        var navbarHeight = this.getNavHeight();
-        this.refs.$BRtoolbar.animate({top: toolbarHeight * -1});
-        this.refs.$BRnav.animate({bottom: navbarHeight * -1});
+        var hide = true;
```

Comment:
Does the value of `hide` ever change or can you just pass `true` to `toggleNavigation`?

Reply:
no, value doesn't change at this point, I just wanted to be very explicit.